packages/react/src/isomorphicClerk.ts

```diff
@@ -356,9 +356,6 @@ export class IsomorphicClerk implements IsomorphicLoadedClerk {
 
   #waitForClerkJS(): Promise<HeadlessBrowserClerk | BrowserClerk> {
     return new Promise<HeadlessBrowserClerk | BrowserClerk>(resolve => {
-      if (this.#loaded) {
-        resolve(this.clerkjs!);
-      }
```

Comment:
❓ Why are we removing this check? Does `addOnLoaded` to the same internally?

Reply:
Exactly! This condition may create another race condition, because this `resolve` may run first, before the ones that are already on the callback array. It's better to always use the callback array so the order is always the same.